rohmu/atomic_opener.py

```diff
@@ -66,7 +65,7 @@ def atomic_opener(
 if TYPE_CHECKING:
```

Comment:
Yes, without this running on Python3.7 fails with:
```
ImportError: cannot import name 'Literal' from 'typing' (/usr/lib64/python3.7/typing.py)
```
(e.g. executing the unittests).

But I think this can be simplified a lot. I believe the current implementation was done to avoid using any type comments... but IMHO if we can use 4 lines instead of 20 with a type comment and when moving to python3.8 simply remove the workaround I think it's better.